From f770f2453a7e4fd4a42d9d952628a70638c0c8fb Mon Sep 17 00:00:00 2001
From: Nava kishore Manne <nava.manne@xilinx.com>
Date: Tue, 29 Sep 2020 13:12:37 +0530
Subject: [PATCH 1698/1852] fpga: zynqmp: Fix incorrect variables type

commit 8fb7cc14bc68ed1469f09d1fdc491133854554b0 from
https://github.com/Xilinx/linux-xlnx.git

Some variables with incorrect type were passed to eemi_ops
API(fpga_read). This API is expecting an "u32 *" but the formal
parameter that was passes was of type "int *". Fixed the issue by
changing the variable types from "int" to "u32".

Addresses-Coverity: "incompatible_param"
Signed-off-by: Nava kishore Manne <nava.manne@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/fpga/zynqmp-fpga.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/fpga/zynqmp-fpga.c b/drivers/fpga/zynqmp-fpga.c
index d549fdc4e0a8..69684fff5c1c 100644
--- a/drivers/fpga/zynqmp-fpga.c
+++ b/drivers/fpga/zynqmp-fpga.c
@@ -280,7 +280,8 @@ static int zynqmp_fpga_read_cfgreg(struct fpga_manager *mgr,
 				   struct seq_file *s)
 {
 	const struct zynqmp_eemi_ops *eemi_ops = zynqmp_pm_get_eemi_ops();
-	int ret = 0, val;
+	int ret = 0;
+	u32 val;
 	unsigned int *buf;
 	dma_addr_t dma_addr = 0;
 	struct zynqmp_configreg *p = cfgreg;
@@ -313,7 +314,8 @@ static int zynqmp_fpga_read_cfgdata(struct fpga_manager *mgr,
 {
 	const struct zynqmp_eemi_ops *eemi_ops = zynqmp_pm_get_eemi_ops();
 	struct zynqmp_fpga_priv *priv;
-	int ret, data_offset;
+	int ret;
+	u32 data_offset;
 	unsigned int *buf;
 	dma_addr_t dma_addr = 0;
 	size_t size;
-- 
2.31.1

